paper.md

```diff
@@ -2,11 +2,77 @@
+
+At this point, one might think "*Why not use a single container for the entire workflow ?*". 
+There are scenarios where a single container image is not suitable for implementing workflows associated to complex application testing or validating scientific explorations 
+[@zheng_integrating_2015]. For example, a workflow might involve 
+executing two steps, both of them requiring conflicting versions of a 
+language runtime (e.g. Python 2.7 **and** Python 3.6). In this 
+scenario, users could resort to solving such conflicts with the use of 
+package managers [@package_managers_wiki], but this defeats the purpose of containers, which is to _not_ have to do this sort of thing inside a container. More 
+generally, the more complex a container image definition gets (a more 
+complex `Dockerfile`), the more "monolithic" it gets, and thus the 
+less maintainable and reusable it is. On the other hand, if an 
+experimentation workflow can be broken down into finer granular 
+units, we end up having pieces of logic that are easier to maintain 
+and reuse.
+
+The above problem has been addressed by several tools in the past in distinct
+ways. Some popular Kubernetes [@kubernetes_google] based workflow engines include Argo [@argocommunity_argoproj_2019] and 
+Pachyderm [@novella_containerbased_2018] which requires access to a fully provisioned Kubernetes cluster. Some stable and generic workflow engines include Taverna [@oinn_taverna_2004] and Pegasus[@deelman_pegasus_2004] which use custom workflow defination languages and assume workflow engine deployment prior to executing workflows.
+
+The main contribution of this paper is the tool called Popper, which follows a container-native [@container_native] strategy for building reproducible worflows from experimental artifacts. We provide a detailed description of the tool in section II. In section III, we present three case studies of how Popper can be used to quickly reproduce complex workflows in different computing environments. We also present a detailed comparison of Popper with existing workflow engines in section V.
```

Comment:
we usually don't reference blog posts in papers. We can define here what container-native means

Reply:
we also want here an actual list of contributions